src/wh_server_keystore.c

```diff
@@ -197,8 +200,9 @@ int hsmReadKey(whServerContext* server, whKeyId keyId, whNvmMetadata* outMeta,
     int i;
     whNvmMetadata meta[1] = {0};
     /* make sure id is valid */
-    if (server == NULL || (keyId & WOLFHSM_KEYID_MASK) == WOLFHSM_KEYID_ERASED
-        || outSz == NULL) {
+    if (server == NULL || ((keyId & WOLFHSM_KEYID_MASK) == WOLFHSM_KEYID_ERASED
+        && (keyId & WOLFHSM_KEYTYPE_MASK) != WOLFHSM_KEYTYPE_SHE) ||
```

Comment:
Does this need to be protected by the compile time ifdef for she_extension?

Reply:
I don't think so, the she keyId stuff is not compile guarded since it's reserved in case they update to use she key later but still want the same keystore